project_stock_product_set/models/product_set_line.py

```diff
@@ -8,19 +8,21 @@ class ProductSetLine(models.Model):
 
     def _prepare_stock_move_values(self, task, quantity):
         self.ensure_one()
-        return {
+        values = {
             "name": self.product_id.display_name,
             "product_id": self.product_id.id,
             "product_uom_qty": self.quantity * quantity,
             "product_uom": self.product_id.uom_id.id,
-            # According to default values set from stock_move field in task form
-            "location_id": (task.location_id.id or task.project_id.location_id.id),
-            "location_dest_id": (
-                task.location_dest_id.id or task.project_id.location_dest_id.id
-            ),
-            "group_id": task.group_id.id,
             "state": "draft",
             "raw_material_task_id": task.id,
-            "picking_type_id": task.picking_type_id.id
-            or task.project_id.picking_type_id.id,
         }
+        # According to default values set from stock_move field in task form
+        stock_move_model = self.env["stock.move"].with_context(
+            default_raw_material_task_id=task.id
+        )
+        values.update(
+            stock_move_model.default_get(
```

Comment:
This doesn't seem correct, as it's not respecting task definition.

Reply:
Ok sorry, I think it is necessary to explain how it was in v15 and how it is now in v16.

In v15 there was the `_set_procurement_group_id()` method (https://github.com/OCA/project/blob/15.0/project_stock/models/project_task.py) in `project_stock` that was used in https://github.com/OCA/project/blob/15.0/project_stock/models/stock_move.py#L90 . In `project_stock_product_set` the method call to set the `group_id` https://github.com/OCA/project/blob/15.0/project_stock_product_set/wizard/product_set_add.py#L33 was used in the wizard.

In v16 there is no `_set_procurement_group_id()` method in `project_stock` and this logic has been defined directly in `default_get()` of `stock.move`: https://github.com/OCA/project/blob/16.0/project_stock/models/stock_move.py#L83 For this reason in `project_stock_product_set` `default_get()` of `stock.move` is called with the `default_raw_material_task_id` context of the corresponding task and so the correct values of: `group_id`, `location_id`, `location_dest_id` and `picking_type_id`.

What do you think would be the appropriate change in this case?

Reply:
According to https://github.com/OCA/project/blob/16.0/project_stock/models/stock_move.py#L89 the definition of the task or the project is respected  for the fields `location_id`, `location_dest_id` and `picking_type_id`, right? What is the reason why this technique does not work in this case?

Reply:
OK, got it. My fault.